src/hist/core.py

```diff
@@ -26,21 +28,58 @@
 ]
 
 
-class BaseHist(Histogram):
+from .axis import Regular
+
+
+class always_normal_method:
+    def __get__(self, instance, owner=None):
+        return partial(self.method, instance or owner())
+
+    def __init__(self, method):
+        self.method = method
+
+
+class BaseHist(bh.Histogram):
     def __init__(self, *args, **kwargs):
         """
             Initialize BaseHist object. Axis params can contain the names.
         """
+        if len(args):
+            self._ax = list(args)
+            super().__init__(*args, **kwargs)
+            self.names: dict = dict()
+            for ax in self.axes:
+                if ax.name in self.names:
+                    raise Exception(
+                        f"{self.__class__.__name__} instance cannot contain axes with duplicated names."
+                    )
+                else:
+                    self.names[ax.name] = True
+        else:
+            self._ax = []
+            self._hist = None
+            self._storage_proxy = None
+
+    @always_normal_method
+    def Regular(self, *args, **kwargs):
+        if self._hist:
+            raise RuntimeError("Cannot add an axis to an existing histogram")
+        self._ax.append(Regular(*args, **kwargs))
+        return self
+
+    def __getattribute__(self, item):
+
+        if not super().__getattribute__("_hist") and not isinstance(
+            super().__getattribute__(item), always_normal_method
```

Comment:
`super()` is also potentially misleading, since the `__getattribute__` is from the class above, but the attribute is from the current class. Usually when you see super you expect the parent behavior. The object version doesn't imply that.

Reply:
But 

```python
if not object.__getattribute__(super(), "_hist") and not isinstance(
            object.__getattribute__(super(), item), always_normal_method
        ):
...
```

will generate


```log
AttributeError: 'super' object has no attribute '_hist'
```

Reply:
`object.__getattribute__(self, item)`

Reply:
The first argument to an unbound method is the instance, not a class. (that's why all method start with "self". So this just says use the object `__getattribute___` method directly.

Reply:
okay.